File: leetcode/src/main/java/com/giwankim/leetcode/support/ListNode.java
package com.giwankim.leetcode.support;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@EqualsAndHashCode
@NoArgsConstructor
public class ListNode {
  public int val;
  public ListNode next;

  public ListNode(int val) {
    this(val, null);
  }

  public static ListNode of(int... values) {
    ListNode head = new ListNode();
    ListNode it = head;
    for (int value : values) {
      it.next = new ListNode(value);
      it = it.next;
    }
    return head.next;
  }

  @Override
  public String toString() {
    return val + "->" + next;
  }
}

File: leetcode-kt/src/main/kotlin/com/giwankim/leetcode/ListNode.kt
package com.giwankim.leetcode

data class ListNode(
    var `val`: Int,
    var next: ListNode? = null,
) {
    companion object {
        @JvmStatic
        fun of(vararg values: Int): ListNode? {
            val head = ListNode(-1)
            var tail: ListNode? = head
            values.forEach { value ->
                tail?.next = ListNode(value)
                tail = tail?.next
            }
            return head.next
        }
    }
}
